node/src/protocol/signature.rs

```diff
@@ -439,6 +437,79 @@ impl SignatureManager {
         messages
     }
 
+    pub fn handle_requests(
+        &mut self,
+        threshold: usize,
+        active: &Participants,
+        my_requests: &mut HashMap<CryptoHash, SignRequest>,
+        presignature_manager: &mut PresignatureManager,
+    ) -> Result<(), super::CryptographicError> {
+        let mut alternate = false;
+        let mut failed_presigs = Vec::new();
+        while let Some(presignature) = {
+            if self.failed_generators.is_empty() && my_requests.is_empty() {
+                None
+            } else {
+                presignature_manager.take_mine()
+            }
+        } {
+            let sig_participants = active.intersection(&[&presignature.participants]);
+            if sig_participants.len() < threshold {
+                tracing::debug!(
+                    participants = ?sig_participants.keys_vec(),
+                    "running: we don't have enough participants to generate a failed signature"
+                );
+                failed_presigs.push(presignature);
+                continue;
+            }
+
+            alternate = !alternate;
+            // NOTE: this prioritizes new requests first then alternates to reattempt failed ones.
+            // TODO: we need to decide how to prioritize certain requests over others such as with gas or
+            // time of when the request made it into the NEAR network. Ticket: https://github.com/near/mpc-recovery/issues/596
+            if alternate {
```

Comment:
if we got no requests in my_requests, but some in failed_generators, since the handle_request fn will always start with alternate = true when entering this if logic, then none of the failed_generators will ever get processed.

If we hope to process only one request/failed_generator on each handle_requests fn call (like we previously had in code, we can do continue when `my_requests.keys().next().cloned()` or `my_requests.remove(&receipt_id)` return None, and break after self.generate().

Reply:
ahh, I think the `break` just ruins this setup. I just got rid of it entirely, but I made it handle failed ones first for now